test_runner/regress/test_timeline_delete.py

```diff
@@ -711,6 +711,7 @@ def test_timeline_delete_works_for_remote_smoke(
     assert tenant_id == env.initial_tenant
     assert main_timeline_id == env.initial_timeline
 
+    assert env.initial_timeline
```

Comment:
This is just bonus, pyright (lsp I use) nags without the assert because the `TimelineId | None` is sticky. Which I guess I could had faked out differently..

Reply:
it's better to have 

```python
assert env.initial_timeline is not None
```

because, currently, it calls `__bool__` method, but potentially, it can be overridden. And again, it's more clear to understand what this assert means.

Reply:
[d230db0](https://github.com/neondatabase/neon/pull/4719/commits/d230db09038dc3b7e7a0eab6f33dcde9a328fc3b)